brigade/plugins/tasks/networking/netmiko_tasks.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import unicode_literals
+
+import os
+
+from brigade.core.task import Result
+
+import clitable
+
+from clitable import CliTableError
+
+from napalm.base.utils import py23_compat
+
+from netmiko import ConnectHandler
+
+
+napalm_to_netmiko_map = {
+    'ios': 'cisco_ios',
+    'nxos': 'cisco_nxos',
+    'eos': 'arista_eos',
+    'junos': 'juniper_junos',
+    'iosxr': 'cisco_iosxr'
+}
+
+
+def get_template_dir():
+    """Find and return the ntc-templates/templates dir."""
+    try:
+        template_dir = os.environ['NET_TEXTFSM']
+        index = os.path.join(template_dir, 'index')
+        if not os.path.isfile(index):
+            # Assume only base ./ntc-templates specified
+            template_dir = os.path.join(template_dir, 'templates')
+    except KeyError:
+        # Construct path ~/ntc-templates/templates
+        home_dir = os.path.expanduser("~")
+        template_dir = os.path.join(home_dir, 'ntc-templates', 'templates')
+
+    index = os.path.join(template_dir, 'index')
+    if not os.path.isdir(template_dir) or not os.path.isfile(index):
+        msg = """
+Valid ntc-templates not found, please install https://github.com/networktocode/ntc-templates
+and then set the NET_TEXTFSM environment variable to point to the ./ntc-templates/templates
+directory."""
+        raise ValueError(msg)
+    return template_dir
+
+
+def get_structured_data(raw_output, platform, command):
+    """Convert raw CLI output to structured data using TextFSM template."""
+    template_dir = get_template_dir()
+    index_file = os.path.join(template_dir, 'index')
+    textfsm_obj = clitable.CliTable(index_file, template_dir)
+    attrs = {'Command': command, 'Platform': platform}
+    try:
+        # Parse output through template
+        textfsm_obj.ParseCmd(raw_output, attrs)
+        return clitable_to_dict(textfsm_obj)
+    except CliTableError:
+        return raw_output
+
+
+def clitable_to_dict(cli_table):
+    """Converts TextFSM cli_table object to list of dictionaries."""
+    objs = []
+    for row in cli_table:
+        temp_dict = {}
+        for index, element in enumerate(row):
+            temp_dict[cli_table.header[index].lower()] = element
+        objs.append(temp_dict)
+    return objs
+
+
+def netmiko_args(task, ip=None, host=None, username=None, password=None, device_type=None,
+                 netmiko_dict=None):
+    """Process arguments coming from different sources."""
+    parameters = {
+        "username": username or task.host["brigade_username"],
+        "password": password or task.host["brigade_password"],
+    }
+
+    if host is None and ip is None:
+        parameters["ip"] = task.host["brigade_ip"]
```

Comment:
I am introducing properties in another branch to simplify dealing with the arguments. Hopefully this will make it easier and more consistent:

https://github.com/napalm-automation/brigade/pull/13/files#diff-3809cfaf6a7bf2bf113debc68df5541cR133

Reply:
Sounds good....will review.